crates/client/src/lib.rs

```diff
@@ -378,6 +379,29 @@ impl<R: RegistryStorage, C: ContentStorage> Client<R, C> {
                 .prove_log_consistency(old_cp.log_root, new_cp.log_root)
                 .await?;
         }
+
+        let operator_records = response.operator;
+        for record in operator_records {
+            let record: ProtoEnvelope<operator::OperatorRecord> = record.try_into()?;
+            let mut operator_info = storage::OperatorInfo::default();
+            if let Err(error) = operator_info.state.validate(&record) {
+                error!("failed to validate: {}", error);
+            }
+            if let Some(head) = operator_info.state.head() {
+                heads.push(LogLeaf {
+                    log_id: LogId::operator_log::<Sha256>(),
+                    record_id: head.digest.clone(),
+                })
+            }
+            self.registry.store_operator(operator_info).await?;
```

Comment:
Storing the operator log state should be moved out of the loop and after proving inclusion.